### PR TITLE
Update cats-effect from 2.5.4 to 3.3.11

### PR DESCRIPTION
**Pull request by olegkovalenko**

### CODE REVIEW

File: src/main/scala/com/evolutiongaming/cluster/pubsub/OptimiseSubscribe.scala [hunk 1]
```diff
@@ -32,15 +32,15 @@ object OptimiseSubscribe {
...
+  def of[F[_] : Async: Parallel]: F[OptimiseSubscribe[F]] = {
     for {
       serialMap <- SerialMap.of[F, String, Subscription[F]]
     } yield {
       apply(serialMap)
     }
   }
 
-  def apply[F[_] : Sync : Parallel](serialMap: SerialMap[F, String, Subscription[F]]): OptimiseSubscribe[F] = {
+  def apply[F[_] : Sync: Parallel](serialMap: SerialMap[F, String, Subscription[F]]): OptimiseSubscribe[F] = {
```
Comment (Z1kkurat) [inline src/main/scala/com/evolutiongaming/cluster/pubsub/OptimiseSubscribe.scala:43]: `Sync` can be replaced with `MonadCancelThrow`

File: src/main/scala/com/evolutiongaming/cluster/pubsub/OptimiseSubscribe.scala [hunk 2]
```diff
@@ -32,15 +32,15 @@ object OptimiseSubscribe {
   type Listener[F[_]] = OnMsg[F, Any]
 
 
-  def of[F[_] : Concurrent : Parallel]: F[OptimiseSubscribe[F]] = {
+  def of[F[_] : Async: Parallel]: F[OptimiseSubscribe[F]] = {
```
Comment (Z1kkurat) [inline src/main/scala/com/evolutiongaming/cluster/pubsub/OptimiseSubscribe.scala:35]: `Async` can be replaced with `Concurrent` + `Runtime`. `Async` is the most powerful typeclass in CE3, combining capabilities from both `Sync` and the rest of the hierarchy, that's why I would prefer to replace `Async` and `Sync` with the least powerful typeclasses when possible in the main code

File: src/main/scala/com/evolutiongaming/cluster/pubsub/OptimiseSubscribe.scala (hunk 1 above)
Comment (olegkovalenko) [reply src/main/scala/com/evolutiongaming/cluster/pubsub/OptimiseSubscribe.scala]: addressed in [3025814](https://github.com/evolution-gaming/pubsub/pull/163/commits/302581413fe6a1632a1ac021e0731f31ec1add31)

File: src/main/scala/com/evolutiongaming/cluster/pubsub/OptimiseSubscribe.scala (hunk 2 above)
Comment (olegkovalenko) [reply src/main/scala/com/evolutiongaming/cluster/pubsub/OptimiseSubscribe.scala]: excellent point, addressed in [6a7aac2](https://github.com/evolution-gaming/pubsub/pull/163/commits/6a7aac2fe8fac4563d05227b068ace36a03f6978)

File: version.sbt [hunk 3]
```diff
@@ -1 +1 @@
-ThisBuild / version := "6.0.7-SNAPSHOT"
+ThisBuild / version := "7.0.0"
```
Comment (Z1kkurat) [inline version.sbt:1]: `7.0.0-SNAPSHOT` please as `7.0.0` will be automatically set by sbt-release during release process

**Comment by Z1kkurat** [comment]
Can you please also drop a couple of lines in README.md saying that the library is based on CE3 starting from 7.0.0 and update the version in the "Setup" section?

File: version.sbt (hunk 3 above)
Comment (olegkovalenko) [reply version.sbt]: good catch, addressed in [9fa088d](https://github.com/evolution-gaming/pubsub/pull/163/commits/9fa088d8f14da6aa79807901770b6dc6eaf24030)

**Comment by olegkovalenko** [comment]
absolutely, addressed in [814a679](https://github.com/evolution-gaming/pubsub/pull/163/commits/814a67950859e35f5dc4eed59b6f07946adea767)

**Review by Z1kkurat — APPROVED** [review]

**Comment by olegkovalenko** [comment]
thank you @Z1kkurat !

**Pull request merged**